pkg/controller/node/node_controller.go

```diff
@@ -973,6 +984,47 @@ func (ctrl *Controller) updateCandidateMachines(pool *mcfgv1.MachineConfigPool,
 	return nil
 }
 
+// setUpdateInPrgressTaint applies in progress taint to all the nodes that are to be updated.
+// The taint on the individual node is removed by MCD once the update of the node is complete.
+// This is to ensure that the updated nodes are being preferred to non-updated nodes there by
+// reducing the number of reschedules.
+func (ctrl *Controller) setUpdateInProgressTaint(nodeName string) error {
+	return clientretry.RetryOnConflict(constants.NodeUpdateBackoff, func() error {
+		oldNode, err := ctrl.kubeClient.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
```

Comment:
nit: can we add a context argument to `setUpdateInProgressTaint` to push this `context.TODO` up a level?  No need to plumb `Context` through legacy call stacks, but new functions that need a context should take a context argument to avoid growing even more tech debt ;).